src/osemgrep/cli_install_semgrep_pro/Install_semgrep_pro_subcommand.ml

```diff
@@ -0,0 +1,218 @@
+(* Brandon Wu
+ *
+ * Copyright (C) 2023 Semgrep Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+open File.Operators
+module Out = Semgrep_output_v1_j
+module Http_helpers = Http_helpers.Make (Lwt_platform)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+(*
+   Parse a semgrep-install-semgrep-pro command, execute it and exit.
+
+   Translated from install.py
+*)
+
+(*****************************************************************************)
+(* Constants *)
+(*****************************************************************************)
+
+let version_stamp_filename = "pro-installed-by.txt"
+
+(*****************************************************************************)
+(* Helpers *)
+(*****************************************************************************)
+
+let add_semgrep_pro_version_stamp current_executable_path =
+  let pro_version_stamp_path =
+    Fpath.parent (Fpath.v current_executable_path) / version_stamp_filename
+  in
+  (* THINK: does this append or write entirely? *)
+  File.write_file pro_version_stamp_path Version.version
+
+let download_semgrep_pro platform_kind dest =
+  let dest = !!dest in
+  match (Semgrep_settings.load ()).api_token with
+  | None ->
+      Logs.err (fun m ->
+          m "No API token found, please run `semgrep login` first.");
+      false
+  | Some token -> (
+      let uri =
+        Uri.(
+          add_query_params'
+            (with_path !Semgrep_envvars.v.semgrep_url
```

Comment:
please move this to Semgrep_app.mli

The goal is what we hide every network access via typed interfaces in Semgrep_App.mli